_inc/lib/debugger/class-jetpack-cxn-test-base.php

```diff
@@ -0,0 +1,295 @@
+				$resolution = __( 'Please ask your hosting provider to confirm your server can make outbound requests to jetpack.com.', 'jetpack' );
+				break;
+			case 'support':
+				$resolution = __( 'Please contact support.', 'jetpack' ); // @todo: Link to support.
+				break;
+		}
+
+		return array(
+			'name'       => $name,
+			'pass'       => false,
+			'message'    => $message,
+			'resolution' => $resolution,
+		);
+	}
+
+	/**
+	 * Provide WP_CLI friendly testing results.
+	 *
+	 * @param string $group Testing group whose results we are outputting. Default "default". Use "all" for all tests.
+	 */
+	public function output_results_for_cli( $group = 'default' ) {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			if ( Jetpack::is_development_mode() ){
+				WP_CLI::line( __( 'Jetpack is in development mode. Please deactivate development mode for accurate results. ') );
+			}
+			WP_CLI::line( __( 'TEST RESULTS:', 'jetpack' ) );
+			foreach ( $this->raw_results( $group ) as $test ) {
+				if ( true === $test['pass'] ) {
+					WP_CLI::log( WP_CLI::colorize( '%gPassed:%n  ' . $test['name'] ) );
+				} elseif ( 'skipped' === $test['pass'] ) {
+					WP_CLI::log( WP_CLI::colorize( '%ySkipped:%n ' . $test['name'] ) );
+					if ( $test['message'] ) {
+						WP_CLI::log( '         ' . $test['message'] ); // Number of spaces to "tab indent" the reason.
+					}
+				} else { // Failed.
+					WP_CLI::log( WP_CLI::colorize( '%rFailed:%n  ' . $test['name'] ) );
+					WP_CLI::log( '         ' . $test['message'] ); // Number of spaces to "tab indent" the reason.
+				}
+			}
+		}
+	}
+
+	/**
+	 * Provide single WP Error instance of all failures.
+	 *
+	 * @param string $group Testing group whose failures we want converted. Default "default". Use "all" for all tests.
+	 *
+	 * @return WP_Error|false WP_Error with all failed tests or false if there were no failures.
+	 */
+	public function output_fails_as_wp_error( $group = 'default' ) {
+		if ( $this->pass( $group ) ) {
+			return false;
+		}
+		$fails = $this->list_fails( $group );
+		$error = false;
+
+		foreach ( $fails as $result ) {
+			$code    = 'failed_' . $result['name'];
+			$message = $result['message'];
+			$data    = array(
+				'resolution' => $result['resolution'],
+			);
+			if ( ! $error ) {
+				$error = new WP_Error( $code, $message, $data );
+			} else {
+				$error->add( $code, $message, $data );
+			}
+		}
+
+		return $error;
+	}
+
+	/**
+	 * Encrypt data for sending to WordPress.com.
+	 *
+	 * @param string $data Data to encrypt with the WP.com Public Key.
+	 *
+	 * @return false|array False if functionality not available. Array of encrypted data, encryption key.
+	 */
+	public function encrypt_string_for_wpcom( $data ) {
+		if ( ! function_exists( 'openssl_get_publickey' ) || ! function_exists( 'openssl_seal' ) ) {
+			return false;
+		}
+
+		$public_key = openssl_get_publickey( JETPACK__DEBUGGER_PUBLIC_KEY );
+
+		if ( openssl_seal( $data, $encrypted_data, $env_key, array( $public_key ) ) ) {
```

Comment:
`openssl_seal()` uses RC4 by default, which is "insecure" (I don't know enough to say whether this use is actually insecure (I think so?) or not).

Different hosts will have different ciphers available. Perhaps we can come up with a list of good ones, look for them in via [openssl_get_cipher_methods()](http://php.net/manual/en/function.openssl-get-cipher-methods.php) (falling back to RC4 as a last resort), and send the cipher/iv in the response so WP.com knows how to decrypt.

We can leave that for another PR, though.

Reply:
I'm inclined to wait on this. The `method` parameter to set the cipher wasn't introduced to PHP until 5.3 and with WordPress moving forward on advancing the minimum supported version of PHP, I think we can add cipher detection and utilizing a stronger one when we're at 5.3+ minimum PHP for Jetpack based on our WP-1 minimum.

Is that reasonable to you?

Reply:
Makes sense.